types/webpack-dev-server/index.d.ts

```diff
@@ -355,24 +276,26 @@ declare module 'webpack' {
 }
 
 declare class WebpackDevServer {
-    listeningApp: WebpackDevServer.ListeningApp;
+    app: express.Application;
+    server: http.Server;
     sockets: NodeJS.EventEmitter[];
 
-    constructor(webpack: webpack.Compiler | webpack.MultiCompiler, config?: WebpackDevServer.Configuration);
+    constructor(config: WebpackDevServer.Configuration, webpack?: webpack.Compiler | webpack.MultiCompiler);
 
-    static addDevServerEntrypoints(
+    static addAdditionalEntries(
         webpackOptions: webpack.Configuration | webpack.Configuration[],
         config: WebpackDevServer.Configuration,
-        listeningApp?: WebpackDevServer.ListeningApp,
     ): void;
 
-    listen(port: number, hostname: string, callback?: (error?: Error) => void): http.Server;
+    start(): void;
+
+    startCallback(callback: () => void): void;
 
-    listen(port: number, callback?: (error?: Error) => void): http.Server;
+    stop(): void;
 
-    close(callback?: () => void): void;
+    stopCallback(callback: () => void): void;
```

Comment:
is this supposed to be unified across module and accept optional `Error | null` param?, say:
https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L1785
https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L1845
https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L1919
https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L1924

Reply:
Honestly, I'm not sure about this, maybe it's legacy from v3?

Reply:
OK, let's go with no params callbacks, as per readme:
https://github.com/webpack/webpack-dev-server/blob/47845fb4f859388249df78af63a37a306f866201/migration-v4.md